### DISCUSSION TITLE
How do you attach an event listener after a native Enum is created?

### DISCUSSION THREAD

**Discussion by sjhewitt** [sqlalchemy/sqlalchemy#8390] [Usage Questions]
I'm trying to execute some custom DDL after the creation of a native Enum, however I can't work out what object to attach the listener to (or if it's even currently possible.)

Minimal code I expected to work:
```python
import enum

from sqlalchemy import create_engine, Column, Enum, Integer, event
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.dialects.postgresql import ENUM
from sqlalchemy.orm import sessionmaker
from sqlalchemy.schema import DDL

db = declarative_base()
db.metadata.schema = 'test'

engine = create_engine('postgresql://postgres:@localhost/test')

class Phase(enum.Enum):
    one = 'one'
    two = 'two'

enum_t = Enum(Phase, name='phase')
# using the postgres dialect ENUM does not work either
# enum_t = ENUM(Phase, name='phase')

class Test(db):
    __tablename__ = 'test'
    id = Column(Integer, primary_key=True)
    phase = Column(enum_t)


event.listen(
    enum_t,
    "after_create",
    DDL(
        """CREATE OR REPLACE FUNCTION some_func(c1 phase)
RETURNS text AS $$
BEGIN
    RETURN (c1::text);
END;
$$ LANGUAGE plpgsql IMMUTABLE STRICT;
"""
    ),
)


db.metadata.create_all(engine)

session = sessionmaker(bind=engine)()
session.execute("select some_func('one'::phase);")
```

Anyone know what I should do?

**Comment by zzzeek** [comment]
mmm I dont have any way for that to work right now, other than adding your event to the table or metadata and having it be idempotent.   the before/after create events only invoke for table and metadata right now.

a comprehensive feature here would need to include them for anything that has a "create", including index, sequence, schematype.   it's worthy of being supported but I'm pretty booked for things to work on in the near to medium term.

**Reply by sjhewitt** [reply]
Thanks for the quick response!

for the moment I've managed to attach to the `after_create` on a table that _should_ always be created before the objects that need access to the created function. It doesn't feel like the perfect solution, but it works for now :)

**Reply by zzzeek** [reply]
this wasn't hard to implement and we will have this in 2.0, see #8394
